diff --git a/sapi/fuzzer/config.m4 b/sapi/fuzzer/config.m4
index 21a44cd6d88..76883258d17 100644
--- a/sapi/fuzzer/config.m4
+++ b/sapi/fuzzer/config.m4
@@ -42,7 +42,7 @@ if test "$PHP_FUZZER" != "no"; then
     [AC_MSG_ERROR([Compiler doesn't support -fsanitize=fuzzer-no-link])])
   ], [
     FUZZING_LIB=$LIB_FUZZING_ENGINE
-    FUZZING_CC="$CXX -stdlib=libc++"
+    FUZZING_CC="$CXX"
   ])
   PHP_SUBST([FUZZING_LIB])
   PHP_SUBST([FUZZING_CC])
diff --git a/sapi/fuzzer/fuzzer-unserialize.c b/sapi/fuzzer/fuzzer-unserialize.c
index 8a889883a97..3ded4651024 100644
--- a/sapi/fuzzer/fuzzer-unserialize.c
+++ b/sapi/fuzzer/fuzzer-unserialize.c
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <stdint.h>
 #include <stdlib.h>
+#include <assert.h>
 
 #include "fuzzer-sapi.h"
 
@@ -66,3 +67,30 @@ int LLVMFuzzerInitialize(int *argc, char ***argv) {
 	/* fuzzer_shutdown_php(); */
 	return 0;
 }
+
+int main(int argc, char **argv) {
+	size_t buflen = 4048;
+	size_t size = 0;
+	uint8_t *buffer = malloc(buflen);
+	assert(buffer != NULL);
+
+	LLVMFuzzerInitialize(&argc, &argv);
+
+	do {
+		size_t readsize = fread(&buffer[size], sizeof(uint8_t), buflen - size, stdin);
+		size += readsize;
+		if (size == buflen) {
+			buflen *= 2;
+			uint8_t *newbuf = realloc(buffer, buflen);
+			if (newbuf == NULL) {
+				free(buffer);
+				return 1;
+			}
+			buffer = newbuf;
+		}
+	} while (!feof(stdin));
+
+	int ret = LLVMFuzzerTestOneInput(buffer, size);
+	free(buffer);
+	return ret;
+}
